Sources/Commands/SwiftRunTool.swift

```diff
@@ -128,9 +163,18 @@ public class SwiftRunTool: SwiftTool<RunToolOptions> {
     }
     
     /// Executes the executable at the specified path.
-    private func run(_ excutablePath: AbsolutePath) throws {
-        let pathRelativeToWorkingDirectory = excutablePath.relative(to: originalWorkingDirectory)
-        try exec(path: excutablePath.asString, args: [pathRelativeToWorkingDirectory.asString] + options.arguments)
+    private func run(_ excutablePath: AbsolutePath, arguments: [String]) throws {
+        let pathRelativeToWorkingDirectory = excutablePath.relative(to: currentWorkingDirectory)
```

Comment:
At that point, `--package-path` will have already changed the working directory, and all paths are relative to `currentWorkingDirectory`, no?